core/src/util/url.cpp

```diff
@@ -214,51 +214,51 @@ Url Url::resolve(const Url& b, const Url& r) {
     // Take the scheme from the base URL.
     if (b.hasScheme()) {
         t.parts.scheme = b.parts.scheme;
-        t.buffer.append(b.buffer, b.parts.scheme.start, b.parts.scheme.count);
+        t.append(b, b.parts.scheme);
         t.buffer.append("://");
     }
 
     // Resolve the netLocation.
-    t.parts.location.start = t.buffer.size();
     if (r.hasNetLocation()) {
         // If the relative URL has a netLocation, use it for the new URL.
-        t.buffer.append(r.buffer, r.parts.location.start, r.parts.location.count);
+        t.parts.location = { t.buffer.size(), r.parts.location.count };
```

Comment:
I had something like this originally, but this initialization syntax isn't supported in GCC 4.9.2 unless you write a constructor for it, so I just set the members separately.